cJSON.c

```diff
@@ -882,10 +888,20 @@ char *cJSON_PrintBuffered(const cJSON *item, int prebuffer, cjbool fmt)
     }
     p.length = prebuffer;
     p.offset = 0;
+    p.noalloc = false;
 
     return print_value(item, 0, fmt, &p);
 }
 
+int cJSON_PrintPreallocated(cJSON *item,char *buf, const int len, const cjbool fmt)
+{
+    printbuffer p;
+    p.buffer = buf;
+    p.length = len;
+    p.offset = 0;
+    p.noalloc = true;
+    return (print_value(item,0,fmt,&p) != NULL ? 0 : -1);
```

Comment:
I meant it exactly the other way around. Think of the return value as a bool. So `return print_value(...) != NULL`. It returns 1 if it succeeds and 0 if it doesn't.